dataverse-webapp/src/main/webapp/dataset.xhtml

```diff
@@ -949,25 +949,18 @@
                     <p class="text-warning">
                         <span class="glyphicon glyphicon-warning-sign"/> #{bundle['dataset.rejectMessage']}
                     </p>
-                    <ui:remove>
-                        <!--FIXME update for new comments table -->
-                        <p:inputTextarea id="returnReason" rows="3" value="#{DatasetPage.workingVersion.returnReason}"
-                                         maxlength="200" widgetVar="returnReason"
-                                         cols="70" counter="display" counterTemplate="{0} characters remaining."
-                                         autoResize="false"/>
-                        <p>
-                            <h:outputText id="display"/>
-                        </p>
-                        <p:watermark for="returnReason" value="#{bundle['dataset.rejectWatermark']}"
-                                     id="returnReasonwatermark"/>
-                    </ui:remove>
+                    <p:inputTextarea id="returnReason" rows="3" value="#{DatasetPage.returnToAuthorReason}"
+                                     maxlength="255" widgetVar="returnReason"
+                                     cols="70" counter="display" counterTemplate="{0} characters remaining."
+                                     autoResize="false"/>
+                    <p>
+                        <h:outputText id="display"/>
+                    </p>
+                    <p:watermark for="returnReason" value="#{bundle['dataset.rejectWatermark']}"
+                                 id="returnReasonwatermark"/>
 
                     <div class="button-block">
-                        <p:commandButton value="#{bundle.continue}" onclick="PF('sendBackToContributor').hide()"
-                                         action="#{DatasetPage.sendBackToContributor}"/>
-                        <ui:remove>
-                            <p:commandButton value="#{bundle.continue}" onclick="testReturnReason();"/>
-                        </ui:remove>
+                        <p:commandButton value="#{bundle.continue}" onclick="testReturnReason();"/>
```

Comment:
testReturnReason function is not safe - you can easily modify javascript on client side.
Which will in the result expose opportunity to execute this action without showing confirm modal window